gateway/ln-gateway/src/lib.rs

```diff
@@ -1851,9 +1854,7 @@ impl Gateway {
             ));
         }
 
-        let contract_amount = payment_info
-            .receive_fee
-            .subtract_fee(payload.invoice_amount.msats);
+        let contract_amount = payment_info.receive_fee.subtract_fee(payload.amount.msats);
```

Comment:
`fee.substract_from(amount)`  could work as a name too, I think.

Reply:
Renamed them according to @dpc's suggestion. Since these are used on the client and on the guardian I do not want to inline. Also saturation is required here.